### PR TITLE
Tooltip: Release v2

### PR DESCRIPTION
**Pull request by diyorbek**
### Summary

- Removed all the Tooltip v2 experiment related code from Popover.
- Deleted all legacy code related to Popover and Tooltip.

### Links

- [Jira](https://jira.pinadmin.com/browse/GWEB-120)
- [TDD](https://docs.google.com/document/d/1GPzNVK9X-A06PgOh5Ha3ncRFb27Cxl0RRBckauFRaZ8/edit?usp=sharing)

### Checklist

- [x] Added unit tests
- [x] Added documentation + accessibility tests
- [x] Verified accessibility: keyboard & screen reader interaction
- [x] Checked dark mode, responsiveness, and right-to-left support
- [x] Checked stakeholder feedback (e.g. Gestalt designers, relevant feature teams)

### CODE REVIEW

File: packages/gestalt/src/Popover/Contents.tsx [hunk 1]
```diff
@@ -51,7 +51,7 @@ export default function Contents({
   hideWhenReferenceHidden,
   onPositioned,
   shouldTrapFocus,
-  overflow = 'auto',
```
Comment (AlbertCarreras) [inline packages/gestalt/src/Popover/Contents.tsx:54]: why are we removing this `auto`?

File: packages/gestalt/src/Popover/Contents.tsx [hunk 2]
```diff
@@ -137,19 +137,16 @@ export default function Contents({
         <div
           aria-label={accessibilityLabel}
           className={classnames(layoutStyles.relative, styles.maxDimensions, {
-            [styles.minDimensions]: width !== null,
+            [styles.minDimensions]: width != null,
```
Comment (AlbertCarreras) [inline packages/gestalt/src/Popover/Contents.tsx:140]: why this change>\?

File: packages/gestalt/src/Popover/InternalPopover.tsx [hunk 3]
```diff
@@ -53,12 +53,12 @@ export default function InternalPopover({
   hideWhenReferenceHidden,
   onPositioned,
   disableFocusTrap = false,
-  overflow,
+  overflow = 'auto',
```
Comment (AlbertCarreras) [inline packages/gestalt/src/Popover/InternalPopover.tsx:56]: oh I see.https://github.com/pinterest/gestalt/pull/3662/files#r1669356514

File: packages/gestalt/src/Popover/InternalPopover.tsx [hunk 4]
```diff
@@ -53,12 +53,12 @@ export default function InternalPopover({
...
 }: Props) {
   const { accessibilityDismissButtonLabel: accessibilityDismissButtonLabelDefault } =
     useDefaultLabelContext('Popover');
 
-  const dismissButtonRef = useRef<null | HTMLAnchorElement | HTMLButtonElement>(null);
```
Comment (AlbertCarreras) [inline packages/gestalt/src/Popover/InternalPopover.tsx:61]: ha thanks

**Review by AlbertCarreras — APPROVED** [review]
LGTM!

File: packages/gestalt/src/Popover/Contents.tsx (hunk 2 above)
Comment (diyorbek) [reply packages/gestalt/src/Popover/Contents.tsx:140]: To remove an incosistency and TS error. `width` is read from `size` prop. `size` accepted both `underifned` and `null`. This creates confusion. If `size` shouldn't be set, then it should be `undefined`, it doesn't have to be `null`.
